eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/Offers.scala

```diff
@@ -135,46 +152,64 @@ object Offers {
     val quantityMin: Option[Long] = records.get[QuantityMin].map(_.min)
     val quantityMax: Option[Long] = records.get[QuantityMax].map(_.max)
 
-    val nodeId: PublicKey = records.get[NodeId].get.nodeId
+    val nodeId: PublicKey = records.get[NodeId].get.nodeIds.head
 
     val sendInvoice: Boolean = records.get[SendInvoice].nonEmpty
 
     val refundFor: Option[ByteVector32] = records.get[RefundFor].map(_.refundedPaymentHash)
 
     val signature: Option[ByteVector64] = records.get[Signature].map(_.signature)
 
-    val contact: OnionMessages.Destination = records.get[Paths].flatMap(_.paths.headOption).map(OnionMessages.BlindedPath).getOrElse(OnionMessages.Recipient(nodeId, None, None))
+    val contact: OnionMessages.Destination = records.get[Paths].flatMap(_.paths.headOption).map(OnionMessages.BlindedPath).getOrElse(OnionMessages.Recipient(nodeId, Some(offerId), None))
```

Comment:
Why do you set a pathId here? We should never set the pathId when we are not the recipient.

Reply:
Good question, I don't remember and this field looks unused, so I don't know how we plan to use it in the future?
Probably worth removing it and adding it only once we're actually using it?